Spring.js

```diff
@@ -158,68 +192,54 @@ const Spring = React.createClass({
     return {
       currVals: endValue,
       currV: mapTree(zero, endValue),
-      now: null,
     };
   },
 
   componentDidMount() {
-    this.raf(true, false);
+    this.startAnimating();
   },
 
   componentWillReceiveProps() {
-    this.raf(true, false);
+    this.startAnimating();
   },
 
   componentWillUnmount() {
-    cancelAnimationFrame(this._rafID);
+    if (this.unsubscribeAnimation) {
+      this.unsubscribeAnimation();
+      this.unsubscribeAnimation = undefined;
+    }
   },
 
-  _rafID: null,
+  startAnimating() {
+    if (!this.unsubscribeAnimation) {
+      this.unsubscribeAnimation = animationLoop.subscribe(
+        this.state, this.animationStep, this.animationRender
+      );
+      animationLoop.start();
+    }
+  },
+
+  animationStep(state, timeStep) {
+    const {currVals, currV} = state;
+    let {endValue} = this.props;
 
-  raf(justStarted, isLastRaf) {
-    if (justStarted && this._rafID != null) {
-      // already rafing
-      return;
+    if (typeof endValue === 'function') {
+      endValue = endValue(currVals);
     }
-    this._rafID = requestAnimationFrame(() => {
-      const {currVals, currV, now} = this.state;
-      let {endValue} = this.props;
 
-      if (typeof endValue === 'function') {
-        endValue = endValue(currVals);
-      }
-      const frameRate = now && !justStarted ? (Date.now() - now) / 1000 : FRAME_RATE;
+    const nextState = updateState(state, endValue, timeStep / 1000);
 
-      const newCurrVals = updateCurrVals(frameRate, currVals, currV, endValue);
-      const newCurrV = updateCurrV(frameRate, currVals, currV, endValue);
+    if (noSpeed(currV) && noSpeed(nextState.currV)) {
+      this.unsubscribeAnimation();
+      this.unsubscribeAnimation = undefined;
```

Comment:
Deleting the property of an object puts the object in “slow mode” under V8 and probably other engines. This impacts performance by quite a bit.